File: Math/Easy/LC_9_Palindrome_Number.cpp
// Problem: https://leetcode.com/problems/palindrome-number/
// Approach: Convert the number to a string and compare symmetric characters.
// Time: O(n) | Space: O(n)

// 9. Palindrome Number
/*
Given an integer x, return true if x is a palindrome, and false otherwise.
*/


class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        string s = to_string(x);
        int i = 0, j = s.size() - 1;
        while (i < j) {
            if (s[i] != s[j]) {
                return false;
            }
            i++; j--;
        }
        return true;
    }
};

File: Math/Easy/LC_9_Palindrome_Number.py
# Problem: https://leetcode.com/problems/palindrome-number/
# Approach: Convert to string and compare with its reverse.
# This directly checks palindrome symmetry.
# Time: O(n) | Space: O(n)

'''
9. Palindrome Number

Given an integer x, return true if x is a palindrome, and false otherwise.
'''




class Solution:
    def isPalindrome(self, x: int) -> bool:
        x = str(x)
        List = x[::]
        if x[::]==x[::-1]:
            return True
        else:
            return False

Solution().isPalindrome(167)
